modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -152,16 +175,140 @@ private BundleDependency createBundleDescriptor(String bundle) {
+                                                         ClassLoaderModelBuilder classLoaderModelBuilder,
+                                                         MulePluginLoaderDescriptor classLoaderModel) {
+    if (!classLoaderModel.getId().equals(MAVEN_ID_CLASSLOADER_MODEL_DESCRIPTOR)) {
+      throw new ArtifactDescriptorCreateException(format("The identifier '%s' for a class loader model descriptor is not supported (error found while reading plugin '%s')",
+                                                         classLoaderModel.getId(), descriptor.getName()));
+    }
+    final File mulePluginPom = new File(pluginFolder, META_INF + separator + MULE_PLUGIN_POM);
+    if (!mulePluginPom.exists()) {
+      throw new ArtifactDescriptorCreateException(format("The identifier '%s' requires the file '%s' (error found while reading plugin '%s')",
+                                                         MAVEN_ID_CLASSLOADER_MODEL_DESCRIPTOR, mulePluginPom.getName(),
+                                                         descriptor.getName()));
+    }
+
+    classLoaderModelBuilder
+        .exportingPackages(new HashSet<>(getAttribute(classLoaderModel, EXPORTED_PACKAGES)))
+        .exportingResources(new HashSet<>(getAttribute(classLoaderModel, EXPORTED_RESOURCES)));
+
+
+    Model model = getPomModel(pluginFolder, mulePluginPom);
+    final Set<BundleDependency> plugins = model.getDependencies().stream()
+        .filter(dependency -> isMulePlugin(dependency))
+        .map(dependency -> {
+          final BundleDescriptor.Builder bundleDescriptorBuilder = new BundleDescriptor.Builder()
+              .setArtifactId(dependency.getArtifactId())
+              .setGroupId(dependency.getGroupId())
+              .setVersion(dependency.getVersion())
+              .setType(dependency.getType());
+          if (isNotBlank(dependency.getClassifier())) {
+            bundleDescriptorBuilder.setClassifier(dependency.getClassifier());
+          }
+
+          return new BundleDependency.Builder()
+              .setDescriptor(bundleDescriptorBuilder.build())
+              .setScope(BundleScope.valueOf(dependency.getScope().toUpperCase()))
+              .build();
+        })
+        .collect(Collectors.toSet());
+    classLoaderModelBuilder.dependingOn(plugins);
+
+    final BundleDescriptor bundleDescriptor = new BundleDescriptor.Builder()
+        .setArtifactId(model.getArtifactId())
+        .setGroupId(model.getGroupId())
+        .setVersion(model.getVersion())
+        .setType(EXTENSION_BUNDLE_TYPE)
+        .setClassifier(MULE_PLUGIN_CLASSIFIER)
+        .build();
+    descriptor.setBundleDescriptor(bundleDescriptor);
+  }
+
+  private boolean isMulePlugin(Dependency dependency) {
+    // We will keep only Mule plugins dependencies or org.mule.runtime.deployment.model.internal.plugin.BundlePluginDependenciesResolver.getArtifactsWithDependencies() will fail looking them up.
```

Comment:
it's how the BundlePluginDependenciesResolver works, which I'm not sure if we should discriminate those deps that are not mule plugins here, or in the tBundlePluginDependenciesResolve class.

I can add a jira to discuss it later